src/service/ConfigLoader.ts

```diff
@@ -371,62 +380,60 @@ export class ConfigLoader {
         return _.map(nodes || [], (node) => {
-            const roles = this.resolveRoles(node);
-            if (node.harvesting && node.api) {
-                return {
-                    syncsource: true,
-                    filespooling: true,
-                    partialtransaction: true,
-                    openPort: true,
-                    sinkType: 'Async',
-                    enableSingleThreadPool: false,
-                    addressextraction: true,
-                    enableAutoSyncCleanup: false,
-                    mongo: true,
-                    zeromq: true,
-                    ...node,
-                    roles,
-                };
-            }
-            if (node.api) {
-                return {
-                    sinkType: 'Async',
-                    syncsource: false,
-                    filespooling: true,
-                    partialtransaction: true,
-                    enableSingleThreadPool: false,
-                    addressextraction: true,
-                    mongo: true,
-                    zeromq: true,
-                    enableAutoSyncCleanup: false,
-                    ...node,
-                    roles,
-                };
-            }
+            const expandedNodeConfiguration = { ...this.getDefaultConfiguration(node), ...node };
+            const roles = this.resolveRoles(expandedNodeConfiguration);
+            return { ...expandedNodeConfiguration, roles };
+        });
+    }
 
-            //peer only (harvesting or not).
+    private getDefaultConfiguration(node: NodePreset): DeepPartial<NodePreset> {
+        if (node.harvesting && node.api) {
             return {
-                sinkType: 'Sync',
-                enableSingleThreadPool: true,
-                addressextraction: false,
-                mongo: false,
-                zeromq: false,
                 syncsource: true,
-                filespooling: false,
-                enableAutoSyncCleanup: true,
-                partialtransaction: false,
-                ...node,
-                roles,
+                filespooling: true,
+                partialtransaction: true,
+                openPort: true,
+                sinkType: 'Async',
+                enableSingleThreadPool: false,
+                addressextraction: true,
+                enableAutoSyncCleanup: false,
+                mongo: true,
+                zeromq: true,
             };
-        });
+        }
+        if (node.api) {
+            return {
+                sinkType: 'Async',
+                syncsource: false,
+                filespooling: true,
+                partialtransaction: true,
+                enableSingleThreadPool: false,
+                addressextraction: true,
+                mongo: true,
+                zeromq: true,
+                enableAutoSyncCleanup: false,
+            };
+        }
+        //peer only (harvesting or not).
+        return {
+            sinkType: 'Sync',
```

Comment:
The default sinkType should be Async

Reply:
Both changes are unrelated to the fix in this PR. 
syncType: Sync and enableSingleThreadPool: true has been there for a long time for peer-only nodes. Are we sure we want to apply this change now?

Reply:
opened https://github.com/nemtech/symbol-bootstrap/issues/217